Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -134,4 +290,51 @@ extension PreplannedMapArea: PreplannedMapAreaProtocol {
     var description: String {
         portalItem.description
     }
+    
+    var id: PortalItem.ID? {
+        portalItem.id
+    }
+}
+
+extension FileManager {
+    private static let mmpkPathExtension: String = "mmpk"
+    private static let offlineMapAreasPath: String = "OfflineMapAreas"
+    private static let packageDirectoryPath: String = "Package"
+    private static let preplannedDirectoryPath: String = "Preplanned"
+    
+    /// The path to the documents folder.
+    private var documentsDirectory: URL {
+        URL.documentsDirectory
+    }
+    
+    /// The path to the offline map areas directory within the documents directory.
+    /// `Documents/OfflineMapAreas`
+    private var offlineMapAreasDirectory: URL {
+        documentsDirectory.appending(
+            path: Self.offlineMapAreasPath,
+            directoryHint: .isDirectory
+        )
+    }
+    
+    /// The path to the web map directory for a specific portal item.
+    /// `Documents/OfflineMapAreas/<Portal Item ID>`
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    private func portalItemDirectory(forPortalItemID portalItemID: Item.ID) -> URL {
```

Comment:
```swift
    private func portalItemDirectory(forPortalItemID portalItemID: PortalItem.ID) -> URL {
```
I prefer to make all of the `Item.ID` become `PortalItem.ID` to improve runtime polymorphism performance, but up to you.